docs/custom-code/prisma-custom-attributes.md

```diff
@@ -0,0 +1,326 @@
+
+## How Amplication generates your database schema with Prisma
+
+First, a quick primer on how Amplication manages and creates your app's database.
+
+Every action that you take in Amplication's UI related to database changes is mapped back to the code that Amplication creates for your project.
+
+For example, let's say you create a new service and choose MySQL as your database of choice.
+Amplication will generate the following Prisma schema:
+
+```
+datasource mysql {
+  provider = "mysql"
+  url      = env("DB_URL")
+}
+
+generator client {
+  provider = "prisma-client-js"
+}
+
+model User {
+  createdAt DateTime @default(now())
+  firstName String?
+  id        String   @id @default(cuid())
+  lastName  String?
+  password  String
+  roles     Json
+  updatedAt DateTime @updatedAt
+  username  String   @unique
+}
+```
+
+:::important
+The _User_ entity is always created by default for every Amplication service.
+:::
+
+The above code snippet is an example of the `schema.prisma` file that Amplication automatically generates for your application.
+Any modifications that you make to your project's _Entities_ and their _Fields_ get reflected as changes in your Prisma schema.
+
+Now let's say you add some new fields to your service's _User_ entity.
+For example, a new `projects` field and a `tasks` field.
+
+After you commit your changes and Amplication builds your project, the newly generated `schema.prisma` file will look like the following:
+
+```
+model User {
+  createdAt DateTime  @default(now())
+  firstName String?
+  id        String    @id @default(cuid())
+  lastName  String?
+  password  String
+  projects  Project[]
+  roles     Json
+  tasks     Task[]
+  updatedAt DateTime  @updatedAt
+  username  String    @unique
+}
+```
+
+:::important
+Go through the [Building New Versions of Your Service](/building-new-versions/) guides you through the steps for adding new entities and fields to your service and committing those changes to your git provider.
+:::
+
+All of the changes that you make through Amplication's UI always translate back to changes in your code, in this case (database changes), to the generated `prisma.schema` file.
+
+**But, Prisma is powerful. It allows you to further refine and customize your schema beyond what Amplication allows you to do purely through the UI.**
+
+**This is why Amplication gives you access to the Custom Attributes text field to fully access that feature and further customize your entities**
+
+## What are Prisma Custom Attributes?
+
+Prisma's Custom Attributes allow you to modify the behavior of an entire Entity or a specific Field.
+
+Let's go through a few examples of when Custom Attributes prove to be useful:
+
+1. **Make sure usernames are unique**: When a user signs up for your application, you want to ensure that each user has a unique username.
+Using Prisma's custom attributes you can enforce this constraint at the field level.
+
+```
+model User {
+  username String @unique
```

Comment:
@dericksozo Unique for a field is supported within the UI. not sure I would mention it as a custom attribute

Reply:
I saw this was already mentioned so this can be ignored.